test/test_brownian.py

```diff
@@ -247,7 +256,54 @@ def test_conditional_statistics(levy_area, use_levy):
             h_std = math.sqrt(h_var)
             normalised_hh = (h_r - h_mean) / h_std
             _, pval_h = stats.kstest(normalised_hh, stats.norm.cdf)
-            assert pval_h > 0.1 / (len(ts) - 2)
+            pvals_h.append(pval_h)
+    return jnp.array(pvals_w1), jnp.array(pvals_w2), jnp.array(pvals_h)
+
+
+@pytest.mark.parametrize("levy_area", ["", "space-time"])
+@pytest.mark.parametrize("use_levy", (False, True))
+def test_conditional_statistics(levy_area: _LA, use_levy):
+    pvals_w1, pvals_w2, pvals_h = conditional_statistics(
+        levy_area, use_levy, tol=2**-8, spacing=2**-10, spline="sqrt"
+    )
+    assert jnp.all(pvals_w1 > 0.1 / pvals_w1.shape[0])
+    if levy_area == "space-time" and use_levy:
+        assert jnp.all(pvals_w2 > 0.1 / pvals_w2.shape[0])
+        assert jnp.all(pvals_h > 0.1 / pvals_h.shape[0])
+
+
+@pytest.mark.parametrize("levy_area", ["", "space-time"])
+def test_spline(levy_area: _LA):
+    if levy_area == "":
+        splines = ("quad", "sqrt", "zero")
+    elif levy_area == "space-time":
+        splines = ("sqrt", "zero")
+    else:
+        assert False
+
+    for spline in splines:
+        pvals_w1, pvals_w2, pvals_h = conditional_statistics(
+            levy_area, use_levy=True, tol=2**-5, spacing=2**-6, spline=spline
+        )
+
+        if spline == "sqrt":
+            # For the correct spline, make sure that all p-values are above
+            # 0.1 (subject to multiple-testing correction) and the average
+            # p-value is above 0.3.
+            def pred(pvals):
+                return jnp.min(pvals) > 0.1 / pvals.shape[0] and jnp.mean(pvals) > 0.3
+
+        else:
+            # make sure that for incorrect splines at least one p-value is
+            # below 0.01 (subject to multiple-testing correction) and the
+            # average p-value is below 0.2.
```

Comment:
I think the average p-value should probably be a lot smaller than this?

Reply:
This is a bit of a subtle issue. So normally yes, it is a lot smaller, exept in one case. When `levy_area="spacetime"` then `pvals_w1` actually are quite good even with `spline="zero"`, due to the randomness the w_r receives from h_su.

What I'm saying is, that the variance that `pval_w1` sees is the variance of the Brownian parabola, but without accounting for the conditioning on H, i.e.
$$
\E[ ( \E[W_{s,r} | W_{s,u}, H_{s,u} ] )^2 - ( \E[W_{s,r} | W_{s,u} ] )^2  ]
$$
Which is actually a very good approximation of the actual variance. In `pval_w2` the influence of H is accounted for in the mean, which is subtracted, and hence the variance can only come from x1 (which is zero when `spline="zero"`). I hope this makes sense.

So what I will do, is to just keep this more permissive upper bound to `pvals_w1` when `levy_area="spacetime"`, and add a stricter upper bound to all the other cases.